pkg/daemon/ceph/osd/daemon.go

```diff
@@ -287,6 +288,32 @@ func matchDevLinks(devLinks, deviceName string) bool {
 	return false
 }
 
+func containBluestoreSignature(device *sys.LocalDisk) (bool, error) {
+	// Old lsblk can't detect an existing OSD.
+	// See: https://github.com/rook/rook/issues/10665
+	logger.Infof("old lsblk can't detect bluestore signature, so try to detect here")
+
+	f, err := os.Open(device.RealPath)
+	if err != nil {
+		err := fmt.Errorf("failed to open %q. %v", device.Name, err)
+		return false, err
+	}
+	defer f.Close()
+	signature := make([]byte, len(bluestoreSignature))
+	_, err = f.Read(signature)
+	if err != nil {
+		err := fmt.Errorf("failed to read %q. %v", device.Name, err)
+		return false, err
+	}
+	if string(signature) == bluestoreSignature {
+		return true, nil
```

Comment:
Can we detect anything besides the signature? For example, the fsid of the cluster or the osd id would be nice to log.

Reply:
modified the PR to print OSD UUID too, thanks.